R/taxonomy.R

```diff
@@ -372,31 +369,59 @@ setMethod("getTaxonomyLabels", signature = c(x = "SummarizedExperiment"),
     #
     all_same_rank <- length(unique(tax_cols_selected)) == 1L
     ans <- mapply("[",
-                  as.data.frame(t(as.data.frame(rd))),
-                  tax_cols_selected,
-                  SIMPLIFY = FALSE)
+                    as.data.frame(t(as.data.frame(rd))),
+                    tax_cols_selected,
+                    SIMPLIFY = FALSE)
     ans <- unlist(ans, use.names = FALSE)
     if(with_rank || !all_same_rank){
         ans <- .add_taxonomic_type(rd, ans, tax_cols_selected)
     }
     ans
 }
 
-#' @rdname taxonomy-methods
-setGeneric("taxonomyTree",
-           signature = "x",
-           function(x, ...)
-               standardGeneric("taxonomyTree"))
+#' Functions for accessing data  
+#' 
+#' These functions find and return data present in a 
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' object.
+#' 
+#' \code{getHierarchyTree} generates a hierarchy tree from the available
+#'   taxonomic information. Internally it uses
+#'   \code{\link[TreeSummarizedExperiment:toTree]{toTree}} and
+#'   \code{\link[TreeSummarizedExperiment:resolveLoop]{resolveLoop}} to sanitize
+#'   data if needed.
+#' 
+#' @inheritParams taxonomy-methods
+#' 
+#' @return
+#' \itemize{
+#'   \item{\code{getHierarchyTree}:} {a \code{phylo} tree representing the 
+#'   hierarchy among available taxonomy information.}
+#' }
+#' 
+#' @name getHierarchyTree
+#' 
+#' @examples
+#' data(GlobalPatterns)
+#' GlobalPatterns
+#' getHierarchyTree(GlobalPatterns)
 
-#' @rdname taxonomy-methods
+#' @rdname get-values
```

Comment:
This name should be hierarchy-tree. Also add the same name above addHierarchyTree -->

1. There should not be get-values page.
2. There should be hiearachy-tree page
3. The title of hiarchy-tree page should be something like ("calculate hireachy tree)
4. *HiearchyTree functions should be removed from taxonomy-methods page.


See man directory. There is manual pages. Check the corresponding pages that they are correct. (In RStudio, you can click the file and then preview.)